samples/tab-meeting-transcript-recording/nodejs/api-server/server.js

```diff
@@ -0,0 +1,227 @@
+const path = require('path');
+const axios = require('axios');
+const { DateTime } = require('luxon');
+const ENV_FILE = path.join(__dirname, '.env');
+const { Readable } = require('stream');
+require('dotenv').config({ path: ENV_FILE });
+
+const clientId = process.env.APP_REGISTRATION_ID;
+const clientSecret = process.env.CLIENT_SECRET;
+const graphScopes = ['https://graph.microsoft.com/User.Read'];
+
+
+//This method obtains an access token and makes requests to the Microsoft Graph API to fetch events, online meetings, transcripts, and recordings.
+app.get('/GetLoginUserInformation', async (req,res) => {
+    const msalClient = new msal.ConfidentialClientApplication({
+        auth: {
+            clientId: clientId,
+            clientSecret: clientSecret
+        }
+    });
+    
+    let tenantId = jwt_decode(req.query.ssoToken)['tid']; //Get the tenant ID from the decoded toke
+
+    msalClient.acquireTokenOnBehalfOf({
+        authority: `https://login.microsoftonline.com/${tenantId}`,
+        oboAssertion: req.query.ssoToken,
+        scopes: graphScopes,
+        skipCache: true
+      })
+      .then( async (result) => {    
+        var result = await getData(result.accessToken)
+        res.send(result);
+      })
+      .catch(error => {
+        console.log("error"+ error.errorCode);
+        res.status(403).json({ error: 'consent_required' });
+    });
+});
+
+// This method is used to fetch meeting transcripts.
+app.get('/getMeetingTranscripts', async (req,res) => {
+    const msalClient = new msal.ConfidentialClientApplication({
+        auth: {
+            clientId: clientId,
+            clientSecret: clientSecret
+        }
+    });
+    
+    let tenantId = jwt_decode(req.query.ssoToken)['tid']; //Get the tenant ID from the decoded toke
+
+    msalClient.acquireTokenOnBehalfOf({
+        authority: `https://login.microsoftonline.com/${tenantId}`,
+        oboAssertion: req.query.ssoToken,
+        scopes: graphScopes,
+        skipCache: true
+      })
+      .then( async (result) => {    
+        const graphApiEndpointOnlineTranscriptsData = `https://graph.microsoft.com/beta/me/onlineMeetings/${req.query.meetingId}/transcripts/${req.query.transcriptId}/content?$format=text/vtt`;
+        const response = await getApiData(graphApiEndpointOnlineTranscriptsData, result.accessToken);
+        res.send(response);
+      })
+      .catch(error => {
+        console.log("error"+ error.errorCode);
+        res.status(403).json({ error: 'consent_required' });
+    });
+});
+
+// This method fetches meeting recordings and passes it as a streamable content
+app.get('/getMeetingRecordings', async (req,res) => {
+    const msalClient = new msal.ConfidentialClientApplication({
+        auth: {
+            clientId: clientId,
+            clientSecret: clientSecret
+        }
+    });
+    
+    let tenantId = jwt_decode(req.query.ssoToken)['tid']; //Get the tenant ID from the decoded toke
+
+    msalClient.acquireTokenOnBehalfOf({
```

Comment:
add this code in common function and use it

Reply:
Fixed